OONIProbeUnitTests/APITest.m

```diff
@@ -68,4 +70,39 @@ -(void)testDownloadUrls{
     }];
 }
 
+- (void)testDeleteJsons {
+    XCTestExpectation *expectation = [self expectationWithDescription:@"testDeleteJsons"];
+    Measurement *existing_1 = [self addMeasurement:EXISTING_REPORT_ID];
+    Measurement *nonexisting_1 = [self addMeasurement:NONEXISTING_REPORT_ID];
+    Measurement *existing_2 = [self addMeasurement:EXISTING_REPORT_ID_2];
+    Measurement *nonexisting_2 = [self addMeasurement:NONEXISTING_REPORT_ID];
+    __block int successes = 0;
+    [TestUtility deleteUploadedJsonsWithMeasurementRemover:^(Measurement *measurement) {
+        if ([measurement.report_id isEqualToString:EXISTING_REPORT_ID] ||
+            [measurement.report_id isEqualToString:EXISTING_REPORT_ID_2]) {
+            @synchronized (self) {
+                successes++;
+                if (successes == 2){
+                    XCTAssert(true);
+                    [expectation fulfill];
+                }
+            }
+        } else {
+            XCTAssert(false);
+            [expectation fulfill];
+        }
+    }];
```

Comment:
I think the test is still incorrect. You need to wait for the expectation. Additionally, fulfilling the expectation inside of the block is wrong. What you want is to count inside the block and verify when the expectation is fulfilled that you have counted correctly. Lastly, you also need to count the cases where unexpected measurements were called, because you want to make sure you operate only on the measurements where you expected to operate.